### PR TITLE
docs: add `isVisible` section and example

### PR DESCRIPTION
**Pull request by AngeloGulina**
In this PR I ty to address [this Issue](https://github.com/vuejs/vue-test-utils-next-docs/issues/47) by:

- explaining what `isVisible()` does
- giving an example where `isVisible()` applies
- explaining why `get()`, `find()` and `find() + exists()` are not suitable

I also modify couple sentences with the intent of:

#### enforcing the understanding that `v-if` removes an element from the DOM (contrary to `v-show`)
In the docs, `Vue` uses `render` in an idiosyncratic way, [for example](https://vuejs.org/v2/guide/conditional.html#v-show) when saying that _elements with `v-show` will always be rendered and remain in the DOM`_.

In browser terms though some elements [hidden via CSS will be omitted from the render tree or laid out as empty boxes](https://developers.google.com/web/fundamentals/performance/critical-rendering-path/render-tree-construction?hl=en).

The documentation updates follow the Vue way

#### enforcing the understanding that `get()` assumes existence and `find()` does not

The new section looks like:

||
|-|
|Screenshot 2020-10-18 at 13 41 14|

### CODE REVIEW

**Comment by AngeloGulina** [comment]
Hey @afontcu, thanks for being flexible and letting me help with this.

Here's a draft. I'm pinging you just to let you know – no urgency.

File: src/guide/conditional-rendering.md [hunk 1]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
...
+Sometimes you only want to hide/show an element while keeping it in the DOM. Vue offers `v-show` for scenarios as such. (You can check the differences between `v-if` and `v-show` [here](https://vuejs.org/v2/guide/conditional.html#v-if-vs-v-show)).
+
+This is how a component with `v-show` looks like:
+
+```js
+const Nav = {
+  template: `
+    <nav>
+      <a id="user" href="/profile">My Profile</a>
+      <ul v-show="shouldShowDropdown" id="user-dropdwn">
```
Comment (AngeloGulina) [inline src/guide/conditional-rendering.md:105]:
```suggestion
      <ul v-show="shouldShowDropdown" id="user-dropdown">
```

File: src/guide/conditional-rendering.md [hunk 2]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
...
+## Using `isVisible()`
+
+`isVisible()` gives you the capacity to check whether an element is hidden/shown by using `v-show`.
+
+```js
+test('does not show the user dropdown', () => {
+  const wrapper = mount(Nav)
+
+  // using `get()` works as well
+  expect(wrapper.find('#user-dropdwn').isVisible()).toBe(false)
```
Comment (AngeloGulina) [inline src/guide/conditional-rendering.md:129]:
```suggestion
  expect(wrapper.find('#user-dropdown').isVisible()).toBe(false)
```

File: src/guide/conditional-rendering.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ Vue Test Utils has a range of features for rendering and making assertions about
 
 ## Finding Elements
 
-One of the most basic features of Vue is the ability to dynamically show, hide and remove elements with `v-if`. Let's look at how to test a component that uses `v-if`.
+One of the most basic features of Vue is the ability to dynamically show, hide by removing elements with `v-if`. Let's look at how to test a component that uses `v-if`.
```
Comment (lmiller1990) [inline src/guide/conditional-rendering.md:7]:
```suggestion
One of the most basic features of Vue is the ability to dynamically insert and remove elements with `v-if`. Let's look at how to test a component that uses `v-if`.
```

File: src/guide/conditional-rendering.md [hunk 4]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
 
 If you have other properties in `data`, don't worry - Vue Test Utils will merge the two together. The `data` in the mounting options will take priority over any default values.
 
-To learn what other mounting options exist, see [`Passing Data`]./passing-data.html) or see [`mounting options`](.,/api/#mount-options).
+To learn what other mounting options exist, see [`Passing Data`](./passing-data.html) or see [`mounting options`](.,/api/#mount-options).
+
+## Checking Elements visibility
+
+Sometimes you only want to hide/show an element while keeping it in the DOM. Vue offers `v-show` for scenarios as such. (You can check the differences between `v-if` and `v-show` [here](https://vuejs.org/v2/guide/conditional.html#v-if-vs-v-show)).
```
Comment (lmiller1990) [inline src/guide/conditional-rendering.md:96]: Can we link to v3 docs?

File: src/guide/conditional-rendering.md [hunk 5]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
...
+  `,
+  data() {
+    return {
+      shouldShowDropdown: false
+    }
+  }
+}
+```
+
+In this scenario, the element is not visible but but it always rendered. `get()` or `find()` will always return a `Wrapper` – `find()` with `.exists()` always return `true` – because the **element is still in the DOM**.
```
Comment (lmiller1990) [inline src/guide/conditional-rendering.md:118]: I wonder if we should mention how this works (display: none) 🤔  maybe too much info... what do you guys think?

**Review by lmiller1990 — COMMENTED** [review]
Nice, very concise, great job

File: src/guide/conditional-rendering.md [hunk 6]
```diff
@@ -50,7 +50,7 @@ If `get()` does not return an element matching the selector, it will raise an er
 
 ## Using `find()` and `exists()`
 
-`get()` works for asserting elements do exist. However, as we mentioned, it throws an error when it can't find an element, so you can't use it to assert whether if elements exist.
+`get()` works on the assumption that elements do exist and throws an error when they do not. It is _not_ recommended to use it for asserting existence.
```
Comment (afontcu) [inline src/guide/conditional-rendering.md:53]: Nice one 👍

File: src/guide/conditional-rendering.md (hunk 5 above)
Comment (afontcu) [inline src/guide/conditional-rendering.md:118]:
```suggestion
In this scenario, the element is not visible but always rendered. `get()` or `find()` will always return a `Wrapper` – `find()` with `.exists()` always return `true` – because the **element is still in the DOM**. 
```

Comment (afontcu) [reply src/guide/conditional-rendering.md:118]: Hm, I'd say we don't need to mention that given that we've already linked to vue docs :)

File: src/guide/conditional-rendering.md [hunk 7]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
...
+    }
+  }
+}
+```
+
+In this scenario, the element is not visible but but it always rendered. `get()` or `find()` will always return a `Wrapper` – `find()` with `.exists()` always return `true` – because the **element is still in the DOM**. 
+
+## Using `isVisible()`
+
+`isVisible()` gives you the capacity to check whether an element is hidden/shown by using `v-show`.
```
Comment (afontcu) [inline src/guide/conditional-rendering.md:122]: I'd suggest rewording this a bit, to make it clear that there is nothing specific to v-show, but rather that it helps you assert against elements that are rendered but not visible, as you explain below. 

I'm afraid reading this sentence in isolation (and we know people tend to skim over docs) might yield the wrong impression.

File: src/guide/conditional-rendering.md [hunk 8]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
...
+
+```js
+test('does not show the user dropdown', () => {
+  const wrapper = mount(Nav)
+
+  // using `get()` works as well
```
Comment (afontcu) [inline src/guide/conditional-rendering.md:128]: good point 🤔  which query should be suggest?

I'd say `get`, as we, using your words, "work on the assumption that the element do exist"

File: src/guide/conditional-rendering.md [hunk 9]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
...
+For any of these cases, `isVisible()` will return `false`.
 
 ## Conclusion
 
-- Use `find()` along with `exists()` to verify whether if an element is in the DOM.
-- Use `get()` if you expect the DOM element to be in the DOM.
+- Use `find()` along with `exists()` to verify whether an element is in the DOM.
+- Use `get()` if you expect the element to be in the DOM.
 - The `data` mounting option can be used to set default values on a component.
+- Use `find()` (or `get()`) with `isVisible()` to verify the visibility of an element that is in the DOM
```
Comment (afontcu) [inline src/guide/conditional-rendering.md:146]: Same as before, I'd say we pick either find or get for these use cases, and suggest them in docs

**Review by afontcu — COMMENTED** [review]
Great addition! 🙌  added some comments here and there to keep the ball rolling 🚀

File: src/guide/conditional-rendering.md (hunk 4 above)
Comment (AngeloGulina) [reply src/guide/conditional-rendering.md:96]:
```suggestion
Sometimes you only want to hide/show an element while keeping it in the DOM. Vue offers `v-show` for scenarios as such. (You can check the differences between `v-if` and `v-show` [here](https://v3.vuejs.org/guide/conditional.html#v-if-vs-v-show)).
```

File: src/guide/conditional-rendering.md (hunk 5 above)
Comment (AngeloGulina) [reply src/guide/conditional-rendering.md:118]: Good point @lmiller1990 and thought about it as well but I decided not to commit the docs to what Vue.js does at the moment (thought it's unlikely to change but... who knows? 😂)

File: src/guide/conditional-rendering.md (hunk 7 above)
Comment (AngeloGulina) [reply src/guide/conditional-rendering.md:122]: Good point, I'll rephrase this!

File: src/guide/conditional-rendering.md [hunk 10]
```diff
@@ -89,10 +89,58 @@ test('renders an admin link', () => {
...
+## Using `isVisible()`
+
+`isVisible()` gives you the capacity to check whether an element is hidden/shown by using `v-show`.
+
+```js
+test('does not show the user dropdown', () => {
+  const wrapper = mount(Nav)
+
+  // using `get()` works as well
+  expect(wrapper.find('#user-dropdown').isVisible()).toBe(false)
```
Comment (AngeloGulina) [inline src/guide/conditional-rendering.md:129]:
```suggestion
  expect(wrapper.get('#user-dropdown').isVisible()).toBe(false)
```

File: src/guide/conditional-rendering.md (hunk 9 above)
Comment (AngeloGulina) [reply src/guide/conditional-rendering.md:146]:
```suggestion
- Use `get()` with `isVisible()` to verify the visibility of an element that is in the DOM
```

File: src/guide/conditional-rendering.md [hunk 11]
```diff
@@ -89,10 +89,57 @@ test('renders an admin link', () => {
...
+    }
+  }
+}
+```
+
+In this scenario, the element is not visible but always rendered. `get()` or `find()` will always return a `Wrapper` – `find()` with `.exists()` always return `true` – because the **element is still in the DOM**. 
+
+## Using `isVisible()`
+
+`isVisible()` gives you the capacity to check whether an element is the DOM is hidden. `isVisible()` will check if:
```
Comment (AngeloGulina) [inline src/guide/conditional-rendering.md:122]: @afontcu is something like this more similar to what you had in mind?

Comment (AngeloGulina) [inline src/guide/conditional-rendering.md:122]:
```suggestion
`isVisible()` gives you the capacity to check whether an element is hidden. `isVisible()` will check if:
```

File: src/guide/conditional-rendering.md [hunk 12]
```diff
@@ -89,10 +89,57 @@ test('renders an admin link', () => {
...
+                                                                                  
+- an element or its ancestors have `display: none`, `visibility: hidden`, `opacity :0` style
+- an element or its ancestors are located inside collapsed `<details>` tag
+- an element or its ancestors have the `hidden` attribute
+
+For any of these cases, `isVisible()` will return `false`.
```
Comment (AngeloGulina) [inline src/guide/conditional-rendering.md:128]:
```suggestion
For any of these cases, `isVisible()` returns `false`.
```

File: src/guide/conditional-rendering.md (hunk 11 above)
Comment (AngeloGulina) [reply src/guide/conditional-rendering.md]: Sorry, this was remove after last minute updates 🙈

Thanks @afontcu and @lmiller1990 for the review. Ready for another round 🏸

Comment (afontcu) [reply src/guide/conditional-rendering.md]: Sounds great, yeah!

**Review by afontcu — APPROVED** [review]
LGTM :) I'll leave it open so Lachlan can review the latest changes.

Thanks for this!! 🙌

**Comment by lmiller1990** [comment]
Looks good. Does this auto deploy now with the netlify stuff? I think so - in that case I'll just merge it and we are done!

Thanks for this contribution... the new docs are easily 10x better than the existing ones. We definitely should back-port some of this stuff if/when we can.

**Comment by AngeloGulina** [comment]
Thanks @afontcu @lmiller1990 🎉  loving contributing to this project.

> We definitely should back-port some of this stuff if/when we can.

If you have an idea of what you'd like to backport, we can open some Issues and I'd be happy to contribute

**Comment by afontcu** [comment]
> Does this auto deploy now with the netlify stuff? I think so

AFAICT, we're not using Netlify here yet, we were waiting to merge VTU-next and VTU-next-docs 😉 Vuepress not supportin Vue 3 was the main stopper.

I'll go ahead and release these changes for now!

Thank you 👐
